src/ContactMessageForm.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_contact_forms;
+
+use Drupal\contact\MessageForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Form controller for contact message forms.
+ *
+ * @internal
+ */
+class ContactMessageForm extends MessageForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+    $form = parent::form($form, $form_state);
+    /** @var \Drupal\contact\MessageInterface $contact_message */
+    $message = $this->entity;
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $message->getContactForm();
+
+    $header = $contact_form->getThirdPartySetting('oe_contact_forms', 'header', FALSE);
+
+    if (!empty($header)) {
+      $form['header'] = [
+        '#markup' => $header,
+      ];
+    }
+
+    // Checkbox to accept privacy policy configured in the ContactForm.
+    $form['privacy_policy'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Privacy policy'),
+      '#description' => $contact_form->getThirdPartySetting('oe_contact_forms', 'privacy_policy', ''),
+      '#required' => TRUE,
+    ];
+
+    // Hide the optional fields the form manager has not configured to be used.
+    $optional_selected = $contact_form->getThirdPartySetting('oe_contact_forms', 'optional_fields', ['oe_country_residence', 'oe_telephone']);
+
+    if (!in_array('oe_country_residence', $optional_selected)) {
+      $form['oe_country_residence']['#access'] = FALSE;
+    }
+
+    if (!in_array('oe_telephone', $optional_selected)) {
+      $form['oe_telephone']['#access'] = FALSE;
+    }
+
+    // Alter the Topics label with the value configured by the form manager.
+    $form['oe_topic']['widget']['#title'] = $contact_form->getThirdPartySetting('oe_contact_forms', 'topic_label', 'Topics');
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function save(array $form, FormStateInterface $form_state): void {
+    /** @var \Drupal\contact\MessageInterface $contact_message */
+    $message = $this->entity;
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $message->getContactForm();
+
+    // If the form is configured to include all the fields in the auto-reply,
+    // set the values after the auto-reply body,
+    // so they get included in the email as well.
+    $reply = $contact_form->getReply();
+    $includes_fields_in_auto_reply = (boolean) $contact_form->getThirdPartySetting('oe_contact_forms', 'includes_fields_in_auto_reply', FALSE);
+
+    if (!empty($reply) && $includes_fields_in_auto_reply === TRUE) {
+      $mail_view = $this->entityTypeManager->getViewBuilder('contact_message')->view($message, 'mail');
+      $reply .= "\n" . \Drupal::service('renderer')->render($mail_view);
```

Comment:
The `$mail_view` is a render array. Let's no render here anything and just add to that render array no so they get rendered later together

Reply:
I dont think its possible, as you can see here https://git.drupalcode.org/project/drupal/-/blob/8.8.x/core/modules/contact/contact.module#L152 no build is called and getReply is a string, so sending an array is not ok.

I've also tried this in hook_mail_alter, appending the build array to body, like so $message['body'][] = $mail_view;
but that throws " Uncaught PHP Exception Exception: "Serialization of 'Closure' is not allowed"

Reply:
ah yes, ok..